src/common/blockchain.sh

```diff
@@ -1,3 +1,201 @@
+
+    echo "Installing fabric contract '$CC_FILE' with id '$CC_ID' and version '$CC_VERSION'..."
+
+    OUTPUT=$(do_curl \
+        -X POST \
+        -u "${BLOCKCHAIN_KEY}:${BLOCKCHAIN_SECRET}" \
+        -F files[]=@"${CC_FILE}" -F chaincode_id="${CC_ID}" -F chaincode_version="${CC_VERSION}" \
+        "${request_url}")
+    
+    if [ $? -eq 1 ]
+    then
+        echo "Failed to install fabric contract:"
+        if [[ "${OUTPUT}" == *"chaincode code"*"exists"* ]]
+        then
+            echo "Chaincode already installed with id '${CC_ID}' and version '${CC_VERSION}'"
+            return 2
+        else
+            echo "Unrecognized error returned:"
+            echo "$OUTPUT"
+            return 1
+        fi  
+    fi
+
+    echo "Successfully installed fabric contract."
+    return 0
+}
+
+function instantiate_fabric_chaincode {
+    CC_ID=$1
+    CC_VERSION=$2
+    CHANNEL=$3
+    INIT_ARGS=$4
+
+    cat << EOF > request.json
+{
+    "chaincode_id": "${CC_ID}",
+    "chaincode_version": "${CC_VERSION}",
+    "chaincode_arguments": [${INIT_ARGS}]
+}
+EOF
+
+    request_url="${BLOCKCHAIN_URL}/api/v1/networks/${BLOCKCHAIN_NETWORK_ID}/channels/${CHANNEL}/chaincode/instantiate"
+
+    echo "Instantiating fabric contract with id '$CC_ID' and version '$CC_VERSION' on channel '$CHANNEL' with arguments '$INIT_ARGS'..."
+
+    OUTPUT=$(
+    do_curl \
+        -X POST \
+        -H 'Content-Type: application/json' \
+        -u "${BLOCKCHAIN_KEY}:${BLOCKCHAIN_SECRET}" \
+        --data-binary @request.json \
+        "${request_url}"
+    )
+    do_curl_status=$?
+
+    rm -f request.json
+
+    if [[ "${OUTPUT}" == *"Failed to establish a backside connection"* || "${OUTPUT}" == *"premature execution"* ]]
+    then
+        echo "Connection problem encountered, delaying 30s and trying again..."
+        sleep 30
+        instantiate_fabric_chaincode "$@"
+        return $?
+    fi
+
+    if [ $do_curl_status -eq 1 ]
+    then
+        echo "Failed to instantiate fabric contract:"
+        if [[ "${OUTPUT}" == *"version already exists for chaincode"* ]]
+        then
+            echo "Chaincode instance already exists with id '${CC_ID}' and version '${CC_VERSION}'"
+            return 2
+        else
+            echo "Unrecognized error returned:"
+            echo "${OUTPUT}"
+            return 1
+        fi
+    fi
+
+    echo "Successfully instantiated fabric contract."
+    return 0
+}
+
+
+function parse_fabric_config {
+    NET_CONFIG_FILE=$1
+
+    echo "Parsing deployment configuration:"
+    cat "$NET_CONFIG_FILE"
+
+    for org in $(jq -r "to_entries[] | .key" "$NET_CONFIG_FILE")
+    do
+        echo "Targeting org '$org'..."
+        authenticate_org "$org"
+        
+        cc_index=0
+        jq -r ".${org}.chaincode[].path" "$NET_CONFIG_FILE" | while read -r CC_PATH
+        do
+            CC_NAME=$(jq -r ".${org}.chaincode[$cc_index].name" "$NET_CONFIG_FILE")
+            CC_FILE="${CC_PATH}/${CC_NAME}.go"
```

Comment:
If I am understanding this correctly you are trying to install a single file.  This is just not going to work.   Each of the directories that we create contains multiple files,  you can't create chaincode using a single .go file, instead each directory will contain several .go files that will make up the complete chaincode logic.   

The API allows you to provide an array of files,  that is what you will have to do in order to send all the files available in the directory.  The other option you have is to create a zip file that contains the content of the directory and send that.  Of course the zip file would have to be created during the build stage.